Sources/TuistGenerator/Utils/SettingsHelper.swift

```diff
@@ -44,4 +38,34 @@ final class SettingsHelper {
     func variant(_ buildConfiguration: BuildConfiguration) -> BuildSettingsProvider.Variant {
         return buildConfiguration.variant == .debug ? .debug : .release
     }
+
+    // MARK: - Private
+
+    private func resolveValue(oldValue: SettingValue?, newValue: SettingValue) -> SettingValue {
+        guard let oldValue = oldValue else {
+            return newValue
+        }
+        guard oldValue != newValue else {
+            return oldValue
+        }
+
+        switch (oldValue, newValue) {
+        case let (.string(old), .string(new)) where new.contains("$(inherited)"):
+            return .array([old, new].uniqued())
+        case let (.string(old), .array(new)) where new.contains("$(inherited)"):
+            return .array(([old] + new).uniqued())
+        case let (.array(old), .string(new)) where new.contains("$(inherited)"):
+            return .array((old + [new]).uniqued())
+        case let (.array(old), .array(new)) where new.contains("$(inherited)"):
+            return .array((old + new).uniqued())
+        default:
+            return newValue
+        }
+    }
+}
+
+private extension Array where Element == String {
```

Comment:
I'd move this to a file where we can define extensions for Array and make it internal.